packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/prompt-helpers.ts

```diff
@@ -157,14 +163,51 @@ export async function connectWithDestination(
  */
 export function getBackendSystemDisplayName(system: BackendSystem): string {
     const userDisplayName = system.userDisplayName ? ` [${system.userDisplayName}]` : '';
-    let systemTypeName = '';
-    if (system.authenticationType === 'reentranceTicket') {
-        systemTypeName = ` (${t('texts.systemTypeS4HC')})`;
+    const systemTypeName = getBackendSystemTypeName(system.systemType);
+    return `${system.name}${systemTypeName}${userDisplayName}`;
+}
+
+/**
+ * Returns the formatted system type name for the given backend system.
+ *
+ * @param systemType the system type to get the name for
+ * @returns system type name formatted as a string, e.g. " (BTP)" or " (S4HC)".
+ */
+function getBackendSystemTypeName(systemType?: string): string {
+    let systemTypeName = ''; // for on prem we do not show the system type
+    if (systemType === 'BTP' || systemType === 'S4HC') {
+        systemTypeName = ` (${systemType})`;
     }
-    if (system.serviceKeys) {
-        systemTypeName = ` (${t('texts.systemTypeBTP')})`;
+    return systemTypeName;
+}
+
+/**
+ * Adds the system type to backend systems for all systems.
+ * Should be removed in a number of sprints.
+ *
+ * @param fileSystemStore - the file system access for backend systems
+ */
+async function addSystemTypeToBackendSystems(fileSystemStore: DataAccess<BackendSystem>): Promise<void> {
```

Comment:
Im not sure odata-service-inquirer is the right place for this work. I think it should be in the store or SAP systems panel.